src/App.tsx

```diff
@@ -1,223 +1,97 @@
-          <div data-cy="Todo" className="todo completed">
-            <label className="todo__status-label">
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
-                defaultChecked
-              />
-            </label>
-
-            <span data-cy="TodoTitle" className="todo__title">HTML</span>
-            <button
-              type="button"
-              className="todo__remove"
-              data-cy="TodoDeleteButton"
-            >
-              ×
-            </button>
-
-            <div data-cy="TodoLoader" className="modal overlay">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
-
-          <div data-cy="Todo" className="todo">
-            <label className="todo__status-label">
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
-              />
-            </label>
+    const getTodosFromServer = async () => {
+      if (user) {
+        try {
+          const todosFromServer = await getTodos(user.id);
 
-            <span data-cy="TodoTitle" className="todo__title">CSS</span>
+          setTodos(todosFromServer);
+        } catch {
+          setHasLoadingError(true);
+          setErrorMessage('Error... Can not load your todos');
+        }
+      }
+    };
 
-            <button
-              type="button"
-              className="todo__remove"
-              data-cy="TodoDeleteButton"
-            >
-              ×
-            </button>
-
-            <div data-cy="TodoLoader" className="modal overlay">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
+    getTodosFromServer();
+  }, []);
 
-          <div data-cy="Todo" className="todo">
-            <label className="todo__status-label">
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
-              />
-            </label>
+  const activeTodos = todos.filter(todo => todo.completed === false);
 
-            <form>
-              <input
-                data-cy="TodoTitleField"
-                type="text"
-                className="todo__title-field"
-                placeholder="Empty todo will be deleted"
-                defaultValue="JS"
-              />
-            </form>
+  let visibleTodos = todos;
 
-            <div data-cy="TodoLoader" className="modal overlay">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
+  switch (filterBy) {
+    case Filter.COMPLETED:
+      visibleTodos = todos.filter(todo => todo.completed === true);
```

Comment:
```suggestion
      visibleTodos = todos.filter(todo => todo.completed);
```

Reply:
also you can move todos.filter outside of switch case and only leave his body, where you add condition, inside